### PR TITLE
8264425: Update building.md on non-English locales on Windows

### PR DESCRIPTION
**Pull request by magicus**
Update the build documentation to reflect the fact that non-English locales are only supported on a "best effort" principle, and add instructions how to install the necessary language pack to Visual Studio.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8264425](https://bugs.openjdk.org/browse/JDK-8264425): Update building.md on non-English locales on Windows (**Bug** - P3)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [6c1cd3b5](https://git.openjdk.org/jdk/pull/16694/files/6c1cd3b535957fe411662dd452fc4ab55c06d874)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [6c1cd3b5](https://git.openjdk.org/jdk/pull/16694/files/6c1cd3b535957fe411662dd452fc4ab55c06d874)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16694/head:pull/16694` \
`$ git checkout pull/16694`

Update a local copy of the PR: \
`$ git checkout pull/16694` \
`$ git pull https://git.openjdk.org/jdk.git pull/16694/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16694`

View PR using the GUI difftool: \
`$ git pr show -t 16694`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16694.diff">https://git.openjdk.org/jdk/pull/16694.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16694#issuecomment-1814833866)

### CODE REVIEW

File: doc/building.md [hunk 1]
```diff
@@ -194,6 +194,22 @@ Linux (WSL), and MSYS2. (MSYS is no longer supported due to an outdated bash;
...
+run correctly. On non-Windows operating systems, this is achieved using the
+`LC_*` variables, which propagate to all child processes of the build.
+Unfortunately, there is no way to set the locale for a specific process like
+this in Windows. The only way is to change the user locale setting to **US
+English**, which will affect all applications run by the user.
+
+Since this is annoying for users who prefer another locale, we strive to get
+the building and testing to work on other locales as well. This is on a "best
+effort" level, so beware! You might get odd results in both building and
+testing. If you do, remember that any other locale than US English are not
```
Comment (erikj79) [inline doc/building.md:207]: Either change `are` -> `is`, or I think even better use suggestion below.
```suggestion
testing. If you do, remember that locales other than US English are not
```

File: doc/building.html [hunk 2]
```diff
@@ -395,6 +395,21 @@ <h3 id="windows">Windows</h3>
 to an outdated bash; While OpenJDK can be built with MSYS2, support for
 it is still experimental, so build failures and unusual errors are not
 uncommon.)</p>
+<p>Building and testing the JDK requires a well-defined locale to be
+guaranteed to run correctly. On non-Windows operating systems, this is
+achieved using the <code>LC_*</code> variables, which propagate to all
+child processes of the build. Unfortunately, there is no way to set the
+locale for a specific process like this in Windows. The only way is to
+change the user locale setting to <strong>US English</strong>, which
+will affect all applications run by the user.</p>
```
Comment (naotoj) [inline doc/building.html:404]: IIUC, it is not only the user locale, but also the system locale that derives the file path encoding should be set to US English (using the `UTF-8` beta option may help though)

Comment (magicus) [reply doc/building.html:404]: Oh, here are lots of news to me.

1) What is the system locale? How do you check which locale is the system locale (as opposed to the user locale), and how to you change it?

2) What is the UTF-8 beta option?

Comment (naotoj) [reply doc/building.html:404]: IMHO, Windows has a history of patching up locale support. They have at least System/User/UI default locales.

That said,
1) System locale is the locale of the OS that affects all users/non-Unicode ANSI applications. You can set it in the ControlPanel->Regional Settings->Administrative->"Change System Locale" button.

2) There is a check box that reads "Beta: Use Unicode UTF-8 for worldwide language support". This enables ANSI applications to use UTF-8 for wide char <-> multibyte conversion. 

So in a nutshell, these combinations of settings end up in many unexpected situations. YMMV

**Comment by magicus** [comment]
The entire markdown file with the changes can be viewed here:
https://github.com/magicus/jdk/blob/readme-locale/doc/building.md#windows

**Review by naotoj — APPROVED** [review]
Thanks for the changes @magicus. This version looks very good in terms of the Windows locale environment.

**Review by erikj79 — APPROVED** [review]
